sdk/src/lib.rs

```diff
@@ -123,33 +121,24 @@ impl ProverClient {
         let proof_id = client.create_proof(elf, &stdin).await?;
         println!("proof_id: {:?}", proof_id);
 
-        let mut pb = StageProgressBar::new();
         loop {
             let (status, maybe_proof) = client
                 .get_proof_status::<BabyBearPoseidon2>(&proof_id)
                 .await?;
 
             match status.status() {
-                ProofStatus::ProofSucceeded => {
+                ProofStatus::ProofFulfilled => {
                     println!("Proof succeeded");
-                    pb.finish();
                     if let Some(proof) = maybe_proof {
                         return Ok(proof);
                     } else {
                         return Err(anyhow::anyhow!("Proof succeeded but no proof available"));
```

Comment:
The logic here is a bit unclear. 

If we fail to fetch the proof with proof status fulfilled, we return an error from the `get_proof_status` function, which then returns another error due to the `?` keyword. If the proof fails to deserialize, we also return an error: `bincode::deserialize(&proof).expect("Failed to deserialize proof")`. 

```rust
let proof = if res.status() == ProofStatus::ProofFulfilled {
            let proof = self
                .http
                .get(res.proof_url.clone())
                .send()
                .await?
                .bytes()
                .await?;
            Some(bincode::deserialize(&proof).expect("Failed to deserialize proof"))
        } else {
            None
        };
```

There doesn't seem to be a state where the error for "proof succeeded but no proof available" occurs, as if the proof is fulfilled, we should always be able to fetch the proof OR we should return an error via `?`.

Given this is the case, we should just unwrap the proof if `ProofFulfilled`, as that's the behavior we expect.

Reply:
added some better error handling to this fn

Reply:
My point here was that we should just do `proof.unwrap()` here. By this point, we can assume that any error from the underlying function was handled with the `?` operators instead of:

```rust
if let Some(proof) = maybe_proof {
    ...
```

Reply:
Sounds good, updated to just do `maybe_proof.unwrap();`